website/static/js/projectorganizer.js

```diff
@@ -1,220 +1,1023 @@
+                    title: $.trim($('#add-folder-input' + theItem.node_id).val())
+                };
+            theItem.expand = false;
+            saveExpandState(theItem, function () {
+                var putAction = $osf.putJSON(url, postData);
+                putAction.done(function () {
+                    //var icon = $('.tb-row[data-id="' + item.id + '"]').find('.tb-toggle-icon'),
+                    //    iconTemplate = treebeard.options.resolveToggle.call(treebeard, item);
+                    //if (icon.get(0)) {
+                    //    m.render(icon.get(0), iconTemplate);
+                    //}
+                    treebeard.updateFolder(null, item);
+                    createBlankProjectDetail();
+                }).fail($osf.handleJSONError);
 
+            });
+            return false;
+        });
+        $('#rename-node-' + theItem.node_id).click(function () {
+            $('#buttons' + theItem.node_id).hide();
+            $('#afc-' + theItem.node_id).hide();
+            $('#findNode' + theItem.node_id).hide();
+            $('#nc-' + theItem.node_id).hide();
+            $('#rnc-' + theItem.node_id).css({'display':'inline-block', 'width' : '100%'});
+        });
+        $('#rename-node-input' + theItem.node_id).bind('keyup', function () {
+            var contents = $.trim($(this).val());
+            if (contents === '' || contents === theItem.name) {
+                $('#rename-node-button' + theItem.node_id).attr('disabled', 'disabled');
+            } else {
+                $('#rename-node-button' + theItem.node_id).removeAttr('disabled');
+            }
+        });
+        $('#rename-node-button' + theItem.node_id).click(function () {
+            var url = theItem.apiURL + 'edit/',
+                postAction,
+                postData = {
+                    name: 'title',
+                    value: $.trim($('#rename-node-input' + theItem.node_id).val())
+                };
+            postAction = $osf.postJSON(url, postData);
+            postAction.done(function () {
+                treebeard.updateFolder(null, theParentNode);
+                createBlankProjectDetail();
+            }).fail($osf.handleJSONError);
+            return false;
         });
+        $('.cancel-button-' + theItem.node_id).click(function () {
+            $('#afc-' + theItem.node_id).hide();
+            $('#rnc-' + theItem.node_id).hide();
+            $('#findNode' + theItem.node_id).hide();
+            $('#nc-' + theItem.node_id).show();
+            $('#buttons' + theItem.node_id).show();
+        });
+        $('#add-item-' + theItem.node_id).click(function () {
+            $('#buttons' + theItem.node_id).hide();
+            $('#afc-' + theItem.node_id).hide();
+            $('#rnc-' + theItem.node_id).hide();
+            $('#findNode' + theItem.node_id).show();
+        });
+    } else {
+        createBlankProjectDetail('Smart folders don\'t have any actions.');
```

Comment:
@caneruguz Why not just show the name of the smart folder like you do with "Dashboard"?